core/dbt/task/deps.py

```diff
@@ -59,40 +121,166 @@ def track_package_install(
             {"name": package_name, "source": source_type, "version": version},
         )
 
-    def run(self) -> None:
-        system.make_directory(self.project.packages_install_path)
+    def check_for_duplicate_packages(self, packages_yml):
+        """Loop through contents of `packages.yml` to ensure no duplicate package names + versions.
+
+        This duplicate check will take into consideration exact match of a package name, as well as
+        a check to see if a package name exists within a name (i.e. a package name inside a git URL).
+
+        Args:
+            packages_yml (dict): In-memory read of `packages.yml` contents
+
+        Returns:
+            dict: Updated or untouched packages_yml contents
+        """
+        for i, pkg_entry in enumerate(packages_yml["packages"]):
+            for val in pkg_entry.values():
+                if self.args.package in val:
+                    del packages_yml["packages"][i]
+
+                    fire_event(DepsFoundDuplicatePackage(removed_package=pkg_entry))
+
+        return packages_yml
+
+    def add(self):
+        packages_yml_filepath = (
+            f"{self.project.project_root}/{self.project.packages_specified_path}"
+        )
+
+        if not system.path_exists(packages_yml_filepath):
+            raise dbt.exceptions.DbtRuntimeError("No packages definition file.")
+
+        if not self.args.package_version and self.args.source != "local":
+            raise dbt.exceptions.DbtRuntimeError(
+                f"Version is required to add a package when source is {self.args.source}"
```

Comment:
nit: this could be a little more actionable of we specified which flag was missing (e.g. --package-version) instead of 'Version'